Stripe/STPPaymentContext.m

```diff
@@ -598,15 +598,14 @@ three_d_secure types of required or optional (not_supported are filtered earlier
                                                                                currency:self.paymentCurrency
                                                                               returnURL:self.configuration.returnURLBlock().absoluteString
                                                                                    card:cardSource.stripeID];
+    threeDSecureParams.metadata = self.sourceInformation.metadata;
 
     STPSourceCompletionBlock threeDSSourceCompletion = ^(STPSource * _Nullable threeDSSource, NSError * _Nullable error) {
         if (error
             || threeDSSource == nil
             || threeDSSource.flow != STPSourceFlowRedirect) {
             if (!threeDSecureRequired
-                && error.domain == StripeDomain
-                && error.code == STPCardError
-                && [error.userInfo[STPCardErrorCodeKey] isEqualToString:STPPaymentMethodNotAvailable]) {
+                && [error.userInfo[STPStripeErrorCodeKey] isEqualToString:@"payment_method_not_available"]) {
```

Comment:
That's fair. Source errors have type `invalid_request`, not `card_error`, so I'm a bit unhappy with putting this under `STPCardErrorCodeKey`, but I think it's probably easier to just do that.